Formula/spidermonkey@91.rb

```diff
@@ -0,0 +1,76 @@
+class SpidermonkeyAT91 < Formula
+  desc "JavaScript-C Engine"
+  homepage "https://spidermonkey.dev"
+  url "https://archive.mozilla.org/pub/firefox/releases/91.6.0esr/source/firefox-91.6.0esr.source.tar.xz"
+  version "91.6.0"
+  sha256 "7e802832152c39588b9a5c8392e90c1b00036bf948fa4a97a7af0d1435ba09a1"
+  license "MPL-2.0"
+
+  # Spidermonkey versions use the same versions as Firefox, so we simply check
+  # Firefox release versions.
+  livecheck do
+    url "https://www.mozilla.org/en-US/firefox/releases/"
+    regex(%r{href=.*?[^\d.](91(?:\.\d+)+)/releasenotes/?["' >]}i)
+  end
+
+  depends_on "autoconf@2.13" => :build
+  depends_on "pkg-config" => :build
+  depends_on "python@3.9" => :build
```

Comment:
Note that I tried `python@3.10` and it failed here with error.
```
==> ../configure --prefix=/usr/local/Cellar/spidermonkey@91/91.6.0 --enable-optimize --enable-readline ...
Last 15 lines from /Users/cho-m/Library/Logs/Homebrew/spidermonkey@91/02.configure:
  File "/private/tmp/spidermonkeyA91-20220210-6205-qg8q9a/firefox-91.6.0/python/mozbuild/mozbuild/util.py", line 1050, in method_call
    cache[args] = self.func(instance, *args)

   ...

  File "/private/tmp/spidermonkeyA91-20220210-6205-qg8q9a/firefox-91.6.0/python/mozbuild/mozbuild/virtualenv.py", line 615, in _disable_pip_outdated_warning
    with open(os.path.join(site_packages, pip_dist_info, "INSTALLER"), "w") as file:
PermissionError: [Errno 1] Operation not permitted: '/usr/local/lib/python3.10/site-packages/pip-22.0.2.dist-info/INSTALLER'
```

Seems like it should be possible to use 3.10, but don't know why it is failing.
On the other hand, `spidermonkey@78` definitely doesn't work with 3.10 unless fixes are backported.